workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts

```diff
@@ -29,45 +29,59 @@ interface TokensExpiration {
   refreshToken: number; // ms
 }
 
-export interface CognitoAuthenticationPluginOptions {
+interface CognitoAppClient {
   /**
-   * The Cognito domain. Follows the format: "https://\<domain prefix\>.auth.\<region\>.amazoncognito.com"
+   * The Cognito app client IDs.
    */
-  cognitoDomain: string;
-
+  clientId: string | string[];
   /**
    * The Cognito user pool ID. Follows the format: "\<region\>_\<some string\>"
    */
   userPoolId: string;
 
   /**
-   * The Cognito app client ID.
+   * The Cognito app client secret.
+   */
+  clientSecret: string;
```

Comment:
Programmatic access clients dont necessarily need client secrets. We just need the client id to add to the JWT verifier

Reply:
Looking through the implementation, I think it would be better to remove it entirely. The client secret is only used for the web UI, the other AppClients wont use it at all

Reply:
It seems to be used for  webUi client in `_getEncodedClientId` method and by extension by `revokeToken`.

Reply:
OK. This means that we won't need `CLIENT_SECRET` variable in `ExampleStack`, correct?

Reply:
The only client secret needed in the example stack is the secret for the web UI app client

Reply:
Removed.